client/state/login/reducer.js

```diff
@@ -45,18 +45,19 @@ export const requestNotice = createReducer( null, {
 	[ TWO_FACTOR_AUTHENTICATION_SEND_SMS_CODE_REQUEST_SUCCESS ]: ( state, { notice } ) => notice
 } );
 
-const updateTwoStepNonce = ( state, { twoStepNonce } ) => Object.assign( {}, state, {
-	two_step_nonce: twoStepNonce
+const updateTwoStepNonce = ( state, { twoStepNonce, nonceType } ) => Object.assign( {}, state, {
+	[ `two_step_nonce_${ nonceType }` ]: twoStepNonce
```

Comment:
Shouldn't we camel case such key since it will be stored in the global state tree?